Tools/CISupport/ews-build/steps.py

```diff
@@ -2255,6 +2282,270 @@ def hideStepIf(self, results, step):
+        self.setProperty('repository', f'{GITHUB_URL}{project}')
+
+        num_prs = yield self.get_number_of_prs_with_label(self.label)
+        if num_prs == 0:
+            yield self._addToLog('stdio', 'Ending process as there are no PRs in safe-merge-queue.\n')
+            return defer.returnValue(SUCCESS)
+        if not num_prs:
+            return defer.returnValue(FAILURE)
+
+        self.setProperty('passed_status_check', [])
+        self.setProperty('failed_status_check', [])
+        self.setProperty('pending_prs', [])
+
+        retrieved_pr_data = yield self.getAllPRData(num_prs)
+        self.build.addStepsAfterCurrentStep([DetermineLabelOwner()])
+
+        defer.returnValue(SUCCESS if retrieved_pr_data else FAILURE)
+
+    def getResultSummary(self):
+        if self.results == SUCCESS:
+            return {'step': f"Successfully retrieved pull request data"}
+        elif self.results == FAILURE:
+            return {'step': f"Failed to retrieve pull request data"}
+        return buildstep.BuildStep.getResultSummary(self)
+
+    @defer.inlineCallbacks
+    def getAllPRData(self, limit, label='\"safe-merge-queue\"'):
+        project = self.getProperty('project') or GITHUB_PROJECTS[0]
+        owner, name = project.split('/', 1)
+        query_body = '{repository(owner:"%s", name:"%s") { pullRequests(labels: %s, last: %s) { edges { node { title number commits(last: 3) { nodes { commit { commitUrl status { state contexts { context state } } } } } } } } } }' % (owner, name, label, limit)
+        query = {'query': query_body}
+
+        yield self._addToLog('stdio', "Fetching all PRs with label {}...\n".format(label))
+
+        response = yield self.query_graph_ql(query)
+        if not response:
+            yield self._addToLog('stderr', 'Failed to retrieve list of PRs.\n')
+            return defer.returnValue(None)
+
+        all_pr_data = response['data']['repository']['pullRequests']['edges']
+        list_of_prs = [pr_data['node']['number'] for pr_data in all_pr_data]
+
+        self.setProperty('list_of_prs', list_of_prs)
+        self.setProperty('all_pr_data', all_pr_data)
+        yield self._addToLog('stdio', 'All PRs in safe-merge-queue: {}\n'.format(list_of_prs))
+        yield self._addToLog('stdio', 'Done!\n')
+
+        defer.returnValue(True)
+
+
+class CheckStatusOfPRsInSafeMergeQueue(buildstep.BuildStep, GitHubMixin, AddToLogMixin):
+    name = 'check-status-of-prs-in-safe-merge-queue'
+    flunkOnFailure = False
+    haltOnFailure = False
+    EMBEDDED_CHECKS = ['ios', 'mac', 'ios-sim', 'ios-wk2', 'ios-wk2-wpt', 'api-ios', 'tv', 'tv-sim', 'watch', 'watch-sim']
+    MACOS_CHECKS = ['mac', 'mac-AS-debug', 'api-mac', 'mac-wk1', 'mac-wk2', 'mac-AS-debug-wk2', 'mac-wk2-stress']
+    QUEUES_FOR_SAFE_MERGE_QUEUE = EMBEDDED_CHECKS + MACOS_CHECKS
+
+    def __init__(self, pr_number='', **kwargs):
+        self.name = f'check-status-of-pr-{pr_number}-in-safe-merge-queue'
```

Comment:
Ditto on basing this on the class variable. Note that this does work in Python, and this is probably the right time to deploy this sort of tactic:
```
template_str = 'abc-{}-123'
resolved_str = template_str.format('def')
```

Reply:
{} aren't allowed to be in Buildbot identifiers and makes unit tests fail. However, I think the step needs to be renamed anyways since it doesn't have to be safe-merge-queue specific, and it's only checking one PR, not multiple PRs. So now it's `CheckStatusOfPR()` with `self.name = '{self.name}-{pr_number}'`